### PR TITLE
cli: Add --max-offset flag and remove MAX_OFFSET env var

### LINKED ISSUE #12878: cli: add --max-offset flag
As discussed in https://github.com/cockroachdb/cockroach/pull/12821, given that users may need to change the max clock offset depending on their cloud environment, we should probably add a `--max-offset` flag to `cockroach start`. This flag would be similar to `--raft-tick-interval` in that it would need to be set identically across all nodes. Eventually, we might want a different way to handle such global configs, but for now, we can just be clear about this in the documentation.

Also, once we've figured out the optimal max offset per cloud provider through our own testing, we can integrate this recommendation into our cloud deployment tutorials. 

@petermattis, @bdarnell

### PR DESCRIPTION
**Pull request by a-robinson**
Fixes #12878

### CODE REVIEW

File: pkg/cli/cliflags/flags.go [hunk 1]
```diff
@@ -324,6 +324,15 @@ disabled by setting the environment variable COCKROACH_SKIP_KEY_PERMISSION_CHECK
 		Description: `Path to the CA key.`,
 	}
 
+	MaxOffset = FlagInfo{
+		Name: "max-offset",
+		Description: `
+Maximum allowed clock offset for the cluster. If observed clock offsets exceed
```
Comment (jseldess) [inline pkg/cli/cliflags/flags.go]: This flag needs to be set identically across all nodes, right? If so, let's add something like this at the end:

> Note that, if this value is changed, every node in the cluster must be stopped and restarted with the identical value.

**Review by jseldess — APPROVED** [review]
LGTM, with one suggestion

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15568#-:-Kj5KciqTeglh7VXMmFu:bnfp4nl)*

File: pkg/cli/cliflags/flags.go (hunk 1 above)
Comment (a-robinson) [inline pkg/cli/cliflags/flags.go]: Done.

And regarding our offline chat about printing the default value, that already gets printed by the CLI machinery by default:

```
$ ./cockroach start --help
[...]
      --max-offset duration
        Maximum allowed clock offset for the cluster. If observed clock offsets
        exceed this limit, servers will crash to minimize the likelihood of
        reading inconsistent data. Increasing this value will increase the time
        to recovery of failures as well as the frequency of uncertainty-based
        read restarts.

        Note that this value must be the same on all nodes in the cluster.
        In order to change it, every node in the cluster must be stopped and
        restarted with the new value.
        (default 500ms)
[...]
`

Comment (jseldess) [inline pkg/cli/cliflags/flags.go]: LGTM. Thanks, @a-robinson.

**Comment by a-robinson** [comment]
cc @cuongdo for the farmer changes I realized I had to make to continue setting the max offset in the backup acceptance test.

**Pull request merged**